proposals/XXXX-stride.md

```diff
@@ -0,0 +1,208 @@
+    //
+}
+```
+
+This proposal makes no provision for offsetting the start or end points of a strided array but you can easily stride an array slice:
+
+```swift
+let results = Array(myArray[5...20].striding(by: 5))
+```
+
+As collections, you can stride both sets and dictionaries. The internal implementation details of sets and dictionaries will return the same strided results each time, although these collections are theoretically unordered. The documentation for `striding(by:)` does not call this out or provide any guarantees of repeatability.
+
+## Detailed Design
+
+```swift
+/// A strided non-contiguous sequence of elements that incorporates
+/// every nth element of a base sequence.
+///
+/// When used with collections, the sequence includes the first element
+/// followed by the element at the `startIndex` offset by the `stride`,
+/// by 2 * the `stride`, etc.
+///
+/// ```
+/// for value in myArray.striding(by: 5) {
+///   // process every 5th member, starting with the
+///   // first member of `myArray`:
+///   //
+///   // myArray[0], myArray[0 + strideLength],
+///   // myArray[0 + 2 * strideLength], ...
+/// }
+/// ```
+///
+/// To stride across a subsequence, use a collection slice:
+/// ```
+/// let results = Array(myArray[5...20].striding(by: 2))
+/// ```
+///
+public struct StridedSequence<BaseSequence: Sequence> : Sequence, IteratorProtocol {
+  public typealias Stride = Int
+  
+  public mutating func next() -> BaseSequence.Element? {
+    defer {
+      for _ in 0 ..< _strideLength - 1 {
+        let _ = _iterator.next()
+      }
+    }
+    return _iterator.next()
+  }
+  
+  /// Access only through `Sequence.striding(by:)`
+  internal init(_ sequence: BaseSequence, stride strideLength: StridedSequence.Stride) {
+    _iterator = sequence.makeIterator()
+    _strideLength = strideLength
+  }
+  
+  internal var _iterator: BaseSequence.Iterator
+  internal var _strideLength: StridedSequence.Stride
+}
+
+extension Sequence {
+  /// Returns a strided iterator of sequence elements. The
+  /// stride length is set to incorporate every nth element.
+  ///
+  /// - Parameter strideLength: the distance for each stride
+  /// - Returns: A strided sequence of values
+  public func striding(by strideLength: StridedSequence<Self>.Stride) -> StridedSequence<Self> {
+    guard strideLength > 0 else { fatalError("Stride must be positive")}
+    return StridedSequence(self, stride: strideLength)
+  }
+}
+```
+
+A more performant implementation is established for `RandomAccessCollection`. This also eliminates trapping for an edge case when striding arrays by their `count`:
+
+```swift
+/// A strided non-contiguous sequence of elements that incorporates
+/// every nth element of a base sequence.
+///
+/// For example:
+///
+/// ```swift
+/// for x in [1, 2, 3, 4, 5, 6].striding(by: 2) {
+///    print(x) // 1, 3, 5
+/// }
+/// ```
+///
+public struct RandomAccessStridedSequence<BaseCollection: RandomAccessCollection> : Collection {
```

Comment:
I recommend you add some rationale for why this can't be `StridedCollection<BaseCollection>`, where `StridedCollection` conforms to the same `Collection` protocols that `BaseCollection` does via conditional conformances to `BidirectionalCollection` and `RandomAccessCollection`.